### PR TITLE
Fix compatibility with transaction >= 2.0 and ZODB >= 5.1.

### PR DESCRIPTION
**Pull request by icemac**

### CODE REVIEW

**Comment by mgedmin** [comment]
Eagerly waiting for a new release to see if it'll fix zodbbrowser's failures!

**Comment by mgedmin** [comment]
Travis failures are from `python setup.py ftest` failing on Python 3 somewhere deep in the internals of setuptools.  I cannot reproduce that failure by running `tox` on my laptop.

Modest proposal: change tox.ini to run zope-testrunner directly.  That's what, e.g., [ZODB's tox.ini](https://github.com/zopefoundation/ZODB/blob/master/tox.ini#L7-L17) does.

**Comment by mgedmin** [comment]
When I said I couldn't reproduce that failure by running `tox` on my laptop, I meant:

- I tried git master and tox ran tests for all Python versions and produced failures because of changes to ZODB
- I tried the `fix_transaction_2` branch and tox ran tests for all Python versions, but tests for all Python 3.x versions failed with a strange `ImportError: No module named 'zope.app.wsgi'` while trying to import `src/zope/app/publication/tests/test_functional.py`.

**Comment by icemac** [comment]
@mgedmin The `ImportError: No module named 'zope.app.wsgi'` we saw too on the computer of @sallner which is running on Linux. It does not happen on my computer (Mac OS). The problem seem to be the following: when importing `zope.app` it points to `.../src/zope/app` instead of pointing to `.../.tox/pyX/lib/pythonX/sitepackages/zope/app`. The latter one is the case on my computer. We used the same versions of Python, pip and tox.

**Comment by icemac** [comment]
@mgedmin Doing tox.ini like ZODB helped for Python 3.4 and 3.5 but 3.3 is still failing.

**Comment by mgedmin** [comment]
Let's drop Python 3.3 support.

**Comment by mgedmin** [comment]
(FWIW that looks like a bug in the `coverage` package to me, but I'm not interested in pursuing it at this time.)

File: tox.ini [hunk 1]
```diff
@@ -3,12 +3,13 @@ envlist = coverage-clean, py27, py33, py34, py35, coverage-report
 
 [testenv]
 commands =
-    coverage run setup.py ftest
+    coverage run --source=zope.app.publication -m zope.testrunner --test-path=src --auto-color --auto-progress
```
Comment (mgedmin) [inline tox.ini]: `--source=zope.app.publication` should not be required, since we specify `source = zope.app.publication` in `./coveragerc`.

**Comment by mgedmin** [comment]
`tox` works for me (including on Python 3.3) on my laptop as well.  It's strange.

Travis has Python 3.3.5, I have Python 3.3.6, but the [changelog](https://hg.python.org/cpython/raw-file/v3.3.6/Misc/NEWS) doesn't show anything that looks relevant.

**Comment by mgedmin** [comment]
I'm tempted to merge this now and figure out what to do with Python 3.3 later.  Dropping support for it seems like the least-painful option.

**Pull request merged**